llm/src/evaluation/eval.py

```diff
@@ -0,0 +1,145 @@
+        results[num] = evaluate(
+                lm=model,
+                task_dict=task_dict,
+                num_fewshot=num,
+                limit=PARTIAL_EVAL_SAMPLE_SIZE if partial_eval_mode else None
+            )
+
+    return results
+
+def log_results_to_tsv(results: JsonResults, outfile: str) -> None:
+    '''
+    Logs the task results to a pandas tsv
+
+    Parameters:
+        results (JsonResults): Model's performance on the tasks.
+        outfile (str): Output file
+    '''
+    dumped = json.dumps(results, indent=2)
+    print(f"Logging results to {outfile}")
+    print(dumped)
+    df = pd.DataFrame(columns=["task", "ppl", "accuracy", "length-normalized_accuracy", "num_fewshot"])
+
+    for num in results:
+        fewshot_exp_results = results[num]['results']
+        for task_name in fewshot_exp_results:
+            accuracy = fewshot_exp_results[task_name].get("acc", "n/a")
+            accuracy_norm = fewshot_exp_results[task_name].get("acc_norm", "n/a")
+            ppl = fewshot_exp_results[task_name].get("ppl", "n/a")
+            df = pd.concat([pd.DataFrame([[task_name, ppl, accuracy, accuracy_norm, num]], columns=df.columns), df], ignore_index=True)
+
+    with open(outfile, "w") as f:
+        df.to_csv(f, sep='\t', index=None)
+
+if __name__ == "__main__":
+    """
+    Example usage from benchmarks/llm directory:
+        python -m src.evaluation.eval --experiment_name opt_1.3b \
+            --model_type pretrained_hf \
+            --model_args checkpoint=facebook/opt-1.3b \
+            --tasks lambada_cloze \
+            --num_fewshot 10
+
+        python -m src.evaluation.eval --experiment_name gpt_neo_125m \
+            --model_type pretrained_hf \
+            --model_args checkpoint=EleutherAI/gpt-neo-125M \
+            --tasks lambada \
+            --num_fewshot 0 1
+
+
+        For composer checkpoints you can either pass in an S3 URI or a local path to a .pt file
+        Your config must also be available locally
+
+        python -m src.evaluation.eval --experiment_name gpt_neo_125m \
+            --model_type composer_checkpoint \
+            --model_args checkpoint=s3://mosaicml-internal-checkpoints-shared/jeremy/gpt-eval/gpt-neo-125m-5rbfZF/checkpoints/ep0-ba80000-rank0.pt \
+                config=yamls/mosaic_gpt/125m.yaml \
+            --tasks lambada \
+            --num_fewshot 0 1 10
+
+
+    """
+    parser = argparse.ArgumentParser(description='Run the EleutherAI eval harness on ComposerGPT models')
+    parser.add_argument('--experiment_name', metavar="experiment_name", type=str)
+    parser.add_argument('--model_type', metavar="model_type", type=str)
+    parser.add_argument('--model_args',  nargs='+', metavar="checkpoint_path", type=str)
+    parser.add_argument('--tasks', metavar="task_name", type=str, nargs='+')
+    parser.add_argument('--num_fewshots', metavar="num_fewshots", type=int, nargs='+')
+    parser.add_argument('--debug', action='store_true', default=False)
```

Comment:
remove

Reply:
`--debug` triggers the pdb trace here: https://github.com/mosaicml/benchmarks/pull/26#discussion_r1029873927
since that was removed, the flag is useless